### PR TITLE
added website_partner_rating module

### PR DESCRIPTION
**Pull request by Amit-Parik**
Added the following features on the module.

User can see reviews details with Reviewer name, Rating, Date, Short & Long Description.User can submit their review only when

If the user is not logged in, then the system will provide link to login page and once after login user will straight away redirected to the partner page.

Option to publish/unpublish reviews from front end website page. For that must be logged in as Administrator. Able to see Avg. Rating and No. of reviews link at the top of partner detail page in website. By clicking on that you will redirect to reviews list.

### CODE REVIEW

**Comment by rafaelbn** [comment]
Please, module must have this template: https://github.com/OCA/maintainer-tools/tree/master/template/module

Read shortly:
- https://odoo-community.org/page/code
- https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md
- https://github.com/OCA/maintainer-tools/blob/master/README.md

Thanks

File: website_partner_rating/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# website_partner_rating
```
Comment (elicoidal) [inline website_partner_rating/README.md]: Hi @Amit-Parik 
Thanks for the contribution. 

Could you use a standard README file, following https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: website_partner_rating/__init__.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline website_partner_rating/__init__.py:1]: Please simplify all headers following https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: website_partner_rating/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#
+##############################################################################
+
+{
+    'name': 'Website Partner Rating',
+    'category': 'Website',
+    'summary': 'Website Partner Rating',
+    'website': 'www.browseinfo.in',
+    'version': '1.0',
+    'description': """
```
Comment (elicoidal) [inline website_partner_rating/__openerp__.py]: no description in manifest => README

File: website_partner_rating/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        You can see reviews details with Reviewer name, Rating, Date, Short & Long Description.User can submit their review only when 
+    they are logged in. If the user is not logged in, then the system will provide link to login page and once after login user will
+    straight away redirected to the partner page.
+        You can publish/unpublish reviews from front end website page. For that you must be logged in as Administrator.
+        You will able to see Avg. Rating and No. of reviews link at the top of partner detail page in website. By clicking on that you will redirect to reviews list.
+
+        """,
+    'author': 'BrowseInfo',
```
Comment (elicoidal) [inline website_partner_rating/__openerp__.py]: Follow "author": "<AUTHOR(S)>, Odoo Community Association (OCA)", and include Odoo Community Association (OCA)

File: website_partner_rating/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+##############################################################################
+
+{
+    'name': 'Website Partner Rating',
+    'category': 'Website',
+    'summary': 'Website Partner Rating',
+    'website': 'www.browseinfo.in',
+    'version': '1.0',
```
Comment (elicoidal) [inline website_partner_rating/__openerp__.py]: Follow the guidelines for the versioning 9.0.1.0.0

File: website_partner_rating/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    'description': """
+        You can see reviews details with Reviewer name, Rating, Date, Short & Long Description.User can submit their review only when
```
Comment (elicoidal) [inline website_partner_rating/__openerp__.py]: space after the period

File: website_partner_rating/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+{
+    'name': 'Website Partner Rating',
+    'category': 'Website',
+    'summary': 'Website Partner Rating',
+    'website': 'www.browseinfo.in',
+    'version': '1.0',
+    'description': """
+        You can see reviews details with Reviewer name, Rating, Date, Short & Long Description.User can submit their review only when 
+    they are logged in. If the user is not logged in, then the system will provide link to login page and once after login user will
```
Comment (elicoidal) [inline website_partner_rating/__openerp__.py]: in general PEP8 for the description as well

File: website_partner_rating/controllers/main.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+from openerp.addons.web import http
+from openerp.addons.web.http import request
+from openerp.addons.website.models.website import slug, unslug
+from openerp.tools.translate import _
+import openerp.http as http
+from openerp.http import request
+
+#from openerp import http, SUPERUSER_ID
+#from openerp.http import request
+#from openerp.tools.translate import _
```
Comment (elicoidal) [inline website_partner_rating/controllers/main.py]: remove dead code

File: website_partner_rating/models/partner.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+    @api.multi
+    def _get_partner_rate( self ):
+        for obj in self:
+            partner_rate = 0.0
+            total_messages = len( [x.id for x in obj.website_message_ids if x.message_rate > 0] )
+            if total_messages > 0:
+                total_rate = sum( [x.message_rate for x in obj.website_message_ids] )
+                getcontext().prec = 3
+                # partner_rate = float(float(total_rate) / float(total_messages))
```
Comment (elicoidal) [inline website_partner_rating/models/partner.py]: remove dead code

File: website_partner_rating/views/partner_rating_view.xml [hunk 10]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (elicoidal) [inline website_partner_rating/views/partner_rating_view.xml]: remove file

File: website_partner_rating/views/template.xml [hunk 11]
```diff
@@ -0,0 +1,419 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- vim:fdn=3:
```
Comment (elicoidal) [inline website_partner_rating/views/template.xml]: remove comment

File: website_partner_rating/views/template.xml [hunk 12]
```diff
@@ -0,0 +1,419 @@
...
+						</div>
+					</div>
+					<p class="post_your_review">Submit your review about this Partner</p>
+
+				</ul>
+
+
+
+
+<!--input t-att-id="'actual-rate-%s' %message.id" t-att-value="message.message_rate"
```
Comment (elicoidal) [inline website_partner_rating/views/template.xml]: remove dead code

File: website_partner_rating/views/template.xml [hunk 13]
```diff
@@ -0,0 +1,419 @@
...
+						<input t-att-id="'rate5-%s' %message.id" type="radio"
+							t-att-name="'rate-%s' %message.id" value="5">
+							<label for="rating5">
+								<span>5</span>
+							</label>
+						</input>
+					</div>
+				</t-->
+
+<hr class="mb32" />
```
Comment (elicoidal) [inline website_partner_rating/views/template.xml]: in general improve indentation

File: website_partner_social_network/README.md [hunk 14]
```diff
@@ -0,0 +1,4 @@
+# partner_extended
```
Comment (elicoidal) [inline website_partner_social_network/README.md]: Normally one module per PR.
Follow guidelines for the README.

File: website_partner_social_network/README.md [hunk 15]
```diff
@@ -0,0 +1,4 @@
...
+
+Adds the social network detail of partner like Facebook,Twitter,Youtube,instagram,Google+
```
Comment (elicoidal) [inline website_partner_social_network/README.md]: add period after commas

File: website_partner_social_network/README.md [hunk 16]
```diff
@@ -0,0 +1,4 @@
...
+and shows this detail on website partner when partner is published
```
Comment (elicoidal) [inline website_partner_social_network/README.md]: Add period

File: website_partner_social_network/__openerp__.py [hunk 17]
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline website_partner_social_network/__openerp__.py]: same remarks as above: headers, description in README, remove empty keys, version number

File: website_partner_social_network/partner_social_network.xml [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+	<data>
+
+        <record id="res_partner_extended_form_view" model="ir.ui.view">
+            <field name="name">res.partner.extended.view.form</field>
+            <field name="model">res.partner</field>
+            <field name="inherit_id" ref="base.view_partner_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//notebook/page[@string='Internal Notes']" position="after">
+                    <page string="Social Networks">
+						<group>
```
Comment (elicoidal) [inline website_partner_social_network/partner_social_network.xml]: indentation

**Comment by rafaelbn** [comment]
Hi @Amit-Parik , are you going to continue this PR? Thanks

**Comment by Amit-Parik** [comment]
Hi,

Oh yes, I forget this to update.

Anyway I will finish it soon.

---

Thanks & regards,
Amit Parik
Contact : +91-8238234631
       (wp) +91-9662885631
amit@browseinfo.in
skype: amit.parik
www.browseinfo.in

**Comment by Amit-Parik** [comment]
Hello,

We have done changes according to your suggestion and imported all files.

Please check.

**Comment by Amit-Parik** [comment]
Hello,

We have done changes according to your suggestion and imported all files.

Please check.

**Comment by rafaelbn** [comment]
Hi @Amit-Parik , thanks! could you please check travis errors?

**Comment by Amit-Parik** [comment]
yes checking.

Thanks

**Comment by Amit-Parik** [comment]
@rafaelbn We have fixed travis errors, please check now.

File: website_partner_rating/controllers/main.py [hunk 19]
```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (simahawk) [inline website_partner_rating/controllers/main.py:2]: comments not attended on headers

File: website_partner_rating/controllers/main.py [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+from openerp.addons.web import http
+from openerp.addons.web.http import request
+import openerp.http as http
+from openerp.http import request
+import openerp.addons.website_crm_partner_assign.controllers.main as website_partner_main
+import werkzeug.urls
+import werkzeug.wrappers
+
+
+class website_partner_rating_comments(website_partner_main.WebsiteCrmPartnerAssign):
```
Comment (simahawk) [inline website_partner_rating/controllers/main.py:31]: use camelcase for classes' names

File: website_partner_rating/controllers/main.py [hunk 21]
```diff
@@ -0,0 +1,55 @@
...
+    """ This method is overloaded for to add messaege_rate and short_description
```
Comment (simahawk) [inline website_partner_rating/controllers/main.py:32]: remove "for"

File: website_partner_rating/controllers/main.py [hunk 22]
```diff
@@ -0,0 +1,55 @@
...
+    in product.template"""
+    @http.route(['/partners/partner/comment/<int:partner_id>'], type='http',
```
Comment (simahawk) [inline website_partner_rating/controllers/main.py:34]: if you are not changing any route param you must remove them all to rely on the default ones: `@http.route()`

File: website_partner_rating/controllers/main.py [hunk 23]
```diff
@@ -0,0 +1,55 @@
...
+                auth="public", methods=['POST'], website=True)
+    def partner_rating(self, partner_id, **post):
```
Comment (simahawk) [inline website_partner_rating/controllers/main.py:36]:
```python
def partner_rating(self, partner_id, comment=None, review=None, **post)
```

File: website_partner_rating/controllers/main.py [hunk 24]
```diff
@@ -0,0 +1,55 @@
...
+        cr, uid, context = request.cr, request.uid, request.context
+        if post.get('comment'):
+            message_id1 = request.registry['res.partner'].message_post(
```
Comment (simahawk) [inline website_partner_rating/controllers/main.py:39]: use `request.env` instead of registry and you'll get a record here, not an ID

Comment (simahawk) [reply website_partner_rating/controllers/main.py:39]: question: are we sure we can write w/ current user?

File: website_partner_rating/controllers/main.py [hunk 25]
```diff
@@ -0,0 +1,55 @@
...
+                cr, uid, partner_id,
+                body=post.get('comment'),
+                type='comment',
+                subtype='mt_comment',
+            context=dict(context))  # mail_create_nosubcribe=True
+            review = post.get('review', 0)
+            short_description = post.get('short_description')
+            cr, uid = request.cr, request.uid
+            mail_message1 = request.registry['mail.message']
+            mail_message1.write(cr, uid, [message_id1], {
```
Comment (simahawk) [inline website_partner_rating/controllers/main.py:49]: w/ the `env` change mentioned above you write on the record straight

File: website_partner_rating/models/partner.py [hunk 26]
```diff
@@ -0,0 +1,73 @@
...
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+##############################################################################
+from openerp import models, fields, api, _
+from decimal import getcontext, Decimal
+
+
+class product_template(models.Model):
```
Comment (simahawk) [inline website_partner_rating/models/partner.py:25]: :fire: isn't this res.partner? :)

File: website_partner_rating/models/partner.py [hunk 27]
```diff
@@ -0,0 +1,73 @@
...
+    _inherit = 'res.partner'
+
+    @api.model
+    def _message_count(self):
```
Comment (simahawk) [inline website_partner_rating/models/partner.py:29]:
```
@api.depends('website_message_ids')
def _compute_message_count(self):
    for item in self:
...
```

File: website_partner_rating/models/partner.py [hunk 28]
```diff
@@ -0,0 +1,73 @@
...
+        self.message_count = self.website_message_ids and len(
+            self.website_message_ids) or 0
+
+    @api.multi
+    def _get_partner_rate(self):
```
Comment (simahawk) [inline website_partner_rating/models/partner.py:34]: _get/_compute

File: website_partner_rating/models/partner.py [hunk 29]
```diff
@@ -0,0 +1,73 @@
...
+        for obj in self:
+            partner_rate = 0.0
+            total_messages = len([x.id for x in obj.website_message_ids if
```
Comment (simahawk) [inline website_partner_rating/models/partner.py:37]: wouldn't be better to use an sql query grouped by partner id or a read_group?

File: website_partner_rating/static/src/css/partner_rating.css [hunk 30]
```diff
@@ -0,0 +1,531 @@
+
+/*----------------------------------------Star Rating Blog----------------------------------------------*/
```
Comment (simahawk) [inline website_partner_rating/static/src/css/partner_rating.css:2]: please use LESS and squash this thing :)

File: website_partner_rating/static/src/css/partner_rating.css [hunk 31]
```diff
@@ -0,0 +1,531 @@
+
```
Comment (simahawk) [inline website_partner_rating/static/src/css/partner_rating.css:1]: blog?

File: website_partner_rating/static/src/js/comment_priority.js [hunk 32]
```diff
@@ -0,0 +1,401 @@
+$("a[href^='#comment']").on("click", function(event) {
```
Comment (simahawk) [inline website_partner_rating/static/src/js/comment_priority.js:1]: encapsulate everything in `odoo.define`

File: website_partner_rating/static/src/js/comment_priority.js [hunk 33]
```diff
@@ -0,0 +1,401 @@
...
+        event.preventDefault();
+        $("html, body").animate({
+            scrollTop: target.offset().top
+        }, 1000);
+    }
+
+});
+var p_id, i1=0, blog_rate_value1;
+$(function() {
+    p_id = document.getElementsByName("product");
```
Comment (simahawk) [inline website_partner_rating/static/src/js/comment_priority.js:15]: well, the whole JS needs some love :)

**Review by simahawk — CHANGES_REQUESTED** [review]
are you still working on this?
otherwise... can we close it?

**Comment by simahawk** [comment]
@Amit-Parik  closing as it's on 9.0 and no action since more than 2 yrs.
Feel free to re-open it, thanks!

**Pull request closed without merging**